searx/external_bang.py

```diff
@@ -0,0 +1,81 @@
+import json
+from os.path import join
+from searx import searx_dir
+
+# bangs data coming from the following url convert to json with
+# https://raw.githubusercontent.com/jivesearch/jivesearch/master/bangs/bangs.toml
+# https://pseitz.github.io/toml-to-json-online-converter/
+# NOTE only use the get_bang_url
+
+
+def get_bang_url(search):
+    """
+    Redirects if the user supplied a correct bang search.
+    :param search: This is a search_query object which contains preferences and the submitted queries.
+    :return: None if the bang was invalid, else a string of the redirect url.
+    """
+
+    if search.external_bang:
+        query = str(search.query, "utf-8")
+        bang = _get_bang(search.external_bang)
+
+        if bang:
+            # TODO add region support.
+            bang_url = bang["regions"]["default"]
+
+            return bang_url.replace("{{{term}}}", _get_bang_query(query))
+    return None
+
+
+def _get_bang(user_bang):
+    """
+    Searches if the supplied user bang is available. Returns None if not found.
+    :param user_bang: The parsed user bang. For example yt
+    :return: Returns a dict with bangs data (check bangs_data.json for the structure)
+    """
+    try:
```

Comment:
Here you can simply write `return _get_bangs_data().get(user_bang)`. `dict().get(key)` returns `None` if the given key doesn't found.

Reply:
true that.

Reply:
Fixed it!